crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1040,27 +1040,53 @@ impl AuthorityPerEpochStore {
         )
     }
 
-    pub fn new_at_next_epoch_for_testing(
+    #[cfg(any(test, feature = "test-utils"))]
+    pub async fn new_at_next_epoch_for_testing(
         &self,
-        backing_package_store: Arc<dyn BackingPackageStore + Send + Sync>,
-        object_store: Arc<dyn ObjectStore + Send + Sync>,
-        expensive_safety_check_config: &ExpensiveSafetyCheckConfig,
+        authority: &Arc<super::AuthorityState>,
     ) -> Arc<Self> {
+        use crate::mock_consensus;
+        use sui_network::randomness;
+
         let next_epoch = self.epoch() + 1;
         let next_committee = Committee::new(
             next_epoch,
             self.committee.voting_rights.iter().cloned().collect(),
         );
-        self.new_at_next_epoch(
+        let new_epoch = self.new_at_next_epoch(
             self.name,
             next_committee,
             self.epoch_start_configuration
                 .new_at_next_epoch_for_testing(),
-            backing_package_store,
-            object_store,
-            expensive_safety_check_config,
+            authority.get_backing_package_store().clone(),
+            authority.get_object_store().clone(),
+            &authority.config.expensive_safety_check_config,
             self.chain_identifier,
+        );
+
+        // Set up randomness manager for the next epoch if randomness is enabled.
```

Comment:
I'm a bit confused at the nature of these changes. Is it a bad merge or some refactoring?

Reply:
Bugfix -- the `reconfigure_for_testing` function (which calls this) would fail any transaction after reconfiguration since it would not include the randomness manager (even if the previous epoch did). This fixes this by setting up the randomness manager for the next epoch if the previous epoch had randomness enabled.

Reply:
Might have been good as a separate PR, I might have been less confused then, but fine as is